### PR TITLE
[Merged by Bors] - feat(Data/Set/Sigma): univ sigma preimage eq self

### PR DESCRIPTION
**Pull request by apnelson1**
This PR adds a single simp lemma stating that for  `α : ι → Type*` and `(s : Set (Σ i, α i))`, we have `univ.sigma (fun i ↦ Sigma.mk i ⁻¹' s) = s`. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
Why do you need this lemma? Can you use `Set.sigma Set.univ f` instead of the union? BTW, could you please add a lemma saying that `Set.sigma` is equal to the indexed union of images?

**Comment by apnelson1** [comment]
> Why do you need this lemma? Can you use `Set.sigma Set.univ f` instead of the union? BTW, could you please add a lemma saying that `Set.sigma` is equal to the indexed union of images?

ah, I hadn't realized the uses of `Set.sigma` for this application. I'll close the PR - thanks!

**Comment by apnelson1** [comment]
> > Why do you need this lemma? Can you use `Set.sigma Set.univ f` instead of the union? BTW, could you please add a lemma saying that `Set.sigma` is equal to the indexed union of images?
> 
> ah, I hadn't realized the uses of `Set.sigma` for this application. I'll close the PR - thanks!

Scratch that - I've reopened! It turns out that the equivalent lemma with better spelling was also missing.

File: Mathlib/Data/Set/Sigma.lean [hunk 1]
```diff
@@ -78,6 +78,10 @@ theorem univ_sigma_univ : (@univ ι).sigma (fun _ ↦ @univ (α i)) = univ := ex
 theorem sigma_univ : s.sigma (fun _ ↦ univ : ∀ i, Set (α i)) = Sigma.fst ⁻¹' s :=
   ext fun _ ↦ and_true_iff _
 
+@[simp] theorem univ_sigma_preimage (s : Set (Σ i, α i)) :
+    (@univ ι).sigma (fun i ↦ Sigma.mk i ⁻¹' s) = s :=
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Sigma.lean:82]: a little more readable as
```suggestion
    (univ : Set ι).sigma (fun i ↦ Sigma.mk i ⁻¹' s) = s :=
```

File: Mathlib/Data/Set/Sigma.lean [hunk 2]
```diff
@@ -78,6 +78,10 @@ theorem univ_sigma_univ : (@univ ι).sigma (fun _ ↦ @univ (α i)) = univ := ex
 theorem sigma_univ : s.sigma (fun _ ↦ univ : ∀ i, Set (α i)) = Sigma.fst ⁻¹' s :=
   ext fun _ ↦ and_true_iff _
 
+@[simp] theorem univ_sigma_preimage (s : Set (Σ i, α i)) :
```
Comment (eric-wieser) [inline Mathlib/Data/Set/Sigma.lean:81]:
```suggestion
@[simp] theorem univ_sigma_preimage_mk (s : Set (Σ i, α i)) :
```

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Comment by YaelDillies** [comment]
@apnelson1 did you trigger the merge? What happened here?

**Comment by apnelson1** [comment]
I replied bors r+ before noticing the actual comments. I'll quickly PR to address them.

**Comment by YaelDillies** [comment]
Ah okay. Over on the reviewers chat we were a bit concerned that the PR had merged itself!

**Comment by apnelson1** [comment]
#15057

**Pull request closed without merging**